pkg/sql/catalog/catalog.go

```diff
@@ -100,3 +102,10 @@ type ResolvedSchema struct {
 	// SchemaUserDefined.
 	Desc SchemaDescriptor
 }
+
+// DescGetter is an interface to retrieve descriptors. In general the interface
```

Comment:
The MapDescGetter and general naming around that stuff moves into this package in a later commit. Please accept the partial measure in this commit.